aws_lambda_powertools/tracing/tracer.py

```diff
@@ -831,7 +876,11 @@ def _disable_xray_trace_batching(self):
         aws_xray_sdk.core.xray_recorder.configure(streaming_threshold=0)
 
     def _is_xray_provider(self):
-        return "aws_xray_sdk" in self.provider.__module__
+        return isinstance(self.provider, XrayProvider)
+
+    def _is_custom_provider(self):
+        # check if provider is not default xray_provider. Avoid test conflits
+        return not self._is_xray_provider() and isinstance(self.provider, BaseProvider)
```

Comment:
I don't understand this code. If I'm passing another provider that is not an instance of the BaseProvider, it will use the XRayProvider, is that correct? Is this intendend?

Overall I would prefer to remove this method, and fix this in the concrete implementation. This would remove a lot of the `# type:ignore` errors that leaves me scared.

Reply:
The reason for this method to exist is to keep the current x-ray tracer and test work as intended. this `_is_custom_provider` is used only to decide if we should call `provider.set_attribute` or the original `provider.put_annotation`/`provider.put_metadata` in Tracer `put_annotation`, `put_metadata` methods. 

The idea is when we introduce custom provider in the future, these two method: `put_annotation`, `put_metadata` should call `provider.set_attribute` instead. But for X-Ray provider and the test dummy provider will get `false` from `_is_custom_provider` and uses current `provider.put_annotation`/`provider.put_metadata`. 

As for your suggestion: " I would prefer to remove this method, and fix this in the concrete implementation" I do agree. To do so, we need to make `provider.set_attribute` to work for both `put_annotation`, `put_metadata` in X-ray sdk and also fix all the impacted test cases.